### ISSUE TITLE
Add fuzzing dictionaries by default

### ISSUE THREAD

**Issue by Google-Autofuzz** [google/oss-fuzz#4334]
Now that Google has a [sizeable amount of fuzzing dictionaries]( https://github.com/google/fuzzing/tree/master/dictionaries ), it would be neat to use them in OSS-Fuzz, either by:

- In the `build.sh` file, per project, git-clone'ing the `fuzzing` repository, and copying the relevant dictionary
- Add an attribute in `project.yaml` that gives type of format and we match a dict automatically

The second way is the most convenient to users, but the first is the easiest. Which one should we pick?

**Comment by alex** [comment]
One benefit of the `project.yaml` approach is that having a common key would make it possible to do smarter cross-polination of corpuses.

One important schema note: it should not be a single attribute per project in project`.yaml`. Some projects, e.g. ImageMagick, have dozens and dozens of formats, and different fuzzers use different ones.

**Comment by inferno-chromium** [comment]
@Alex just fyi, clusterfuzz already cross pollinate corpus from other fuzz targets in a random fashion every day. Like once a day when it minimizes corpus, it also cross pollinates them.

**Comment by alex** [comment]
@inferno-chromium right, presumabely having metadata like "this is a PNG and that's a PNG fuzzer too" would let you do smarter cross-polinization. I assume right now you do a lot of "try to pass x.509 corpus to a TIFF fuzzer" which isn't valuable. (But maybe Google just has enough computers that it works fine anyways)

**Comment by inferno-chromium** [comment]
> @inferno-chromium right, presumabely having metadata like "this is a PNG and that's a PNG fuzzer too" would let you do smarter cross-polinization. I assume right now you do a lot of "try to pass x.509 corpus to a TIFF fuzzer" which isn't valuable. (But maybe Google just has enough computers that it works fine anyways)

Actually there is a way to tag things too, @mbarbella-chromium - can you explain more. i think we need to productionize that and hook things into its project.yaml or somewhere in fuzzer options file where people can add that tag [instead of current way of admins adding tag in some place in db]

**Comment by Dor1s** [comment]
@Google-Autofuzz as we discussed on chat, `project.yaml` cannot be used for that, as it has provides project level controls and metadata, not a fuzz target level.

And in general I'm not supportive of automated use of manually compiled dictionaries. We have support for recommended dictionaries, which are distilled and supposed to have only "useful" elements, while manually compiled dictionaries are constant and may have irrelevant elements.
